polyfill/lib/plaindate.mjs

```diff
@@ -133,8 +133,8 @@ export class PlainDate {
     if (!props) {
       throw new TypeError('invalid date-like');
     }
-    const fields = ES.ToTemporalDateFields(this, fieldNames);
-    ObjectAssign(fields, props);
+    let fields = ES.ToTemporalDateFields(this, fieldNames);
```

Comment:
This already existed before this change: `plainDate.with({ day: 15 })` calls the `day` getter on the receiver, even though the value is overwritten by 15.

I think having the Calendar method return a list of fields that should be deleted from the fields of the Temporal object is a valid way of achieving the same thing, but it's my opinion that mergeFields() is an easier operation to reason about.

I don't think we should pass the Temporal object through to the calendar, because then mergeFields() would have to Get different fields depending on the type of the Temporal object.

Reply:
Seems reasonable. I don't have a strong opinion either way.